operator/v1/types_csi_cluster_driver.go

```diff
@@ -96,6 +96,44 @@ type ClusterCSIDriverSpec struct {
 	// The current default behaviour is Managed.
 	// +optional
 	StorageClassState StorageClassStateName `json:"storageClassState,omitempty"`
+
+	// DriverConfig field can be used to optionally specify
+	// driver specific configuration.
+	// If this field is nil, platform chosen default configuration
+	// will be applied.
+	// +kubebuilder:validation:Optional
+	// +optional
+	DriverConfig *CSIDriverConfigSpec `json:"driverConfig,omitempty"`
+}
+
+// CSIDriverConfigSpec defines configuration spec that can be
+// used to optionally configure a specific CSI Driver.
+// +union
+type CSIDriverConfigSpec struct {
+	// DriverName indicates which of the driver we are
+	// installing
+	// +kubebuilder:validation:Required
+	// +unionDiscriminator
+	// +required
+	DriverName CSIDriverName `json:"driverName"`
+
+	// VSphere field is used to configure vsphere CSI driver
+	// specific properties.
+	// +kubebuilder:validation:Optional
+	// +optional
+	VSphere *VSphereCSIDriverConfigSpec `json:"vsphere,omitempty"`
+}
+
+// VSphereCSIDriverConfigSpec defines properties that
+// can be configured for vsphere CSI driver.
+type VSphereCSIDriverConfigSpec struct {
+	// TopologyCategories indicates categories with which
+	// vcenter resources as hostcluster or datacenter were tagged with.
```

Comment:
Can you clarify what this means? It doesn't make sense to me, do you mean `resources such as`?

Reply:
Added more details. PTAL